linkedin/tools/users.py

```diff
@@ -1,8 +1,30 @@
-from tools.helper import tool_registry, ACCESS_TOKEN
+from tools.helper import ACCESS_TOKEN
 from linkedin_api.clients.restli.client import RestliClient
+import time
+import sys
 
 
-@tool_registry.register_tool("GetCurrentUser")
-def get_user(client: RestliClient):
-    response = client.get(resource_path="/userinfo", access_token=ACCESS_TOKEN)
+def get_user(client: RestliClient) -> dict:
+
+    MAX_RETRIES = 3
+    DELAY = 1  # seconds between retries
+
+    for attempt in range(1, MAX_RETRIES + 1):
```

Comment:
It seems strange to me that we would need to retry this, Especially for a 401 error.

Reply:
yeah, I am totally with you. Here is the detail of the issue:
Sometimes, the first time the tool makes a `get_user()` request , it gets such error response back:
```
\"status\": 401, \"serviceerrorcode\": 65601, \"code\": \"revoked_access_token\
```
and then the second time it just works without changing anything.
==================
My theory is: sometimes, the certain part of the Oauth process in the LinkedIn API server is so slow, such that after the access token is granted, our `get_user()` request hits the server earlier than the moment the access token is marked as a valid one in their system.

So I did find another way to fix this : add something like `time.sleep(2)` to the beginning of this function, and it works ..But  a `retry` probably makes more sense.